src/Umbraco.Core/Services/DataTypeService.cs

```diff
@@ -246,6 +248,47 @@ public Task<IEnumerable<IDataType>> GetAllAsync(params Guid[] keys)
             return Task.FromResult<IEnumerable<IDataType>>(dataTypes);
         }
 
+        /// <inheritdoc />
+        public async Task<IEnumerable<IDataType>> FilterAsync(string orderBy = "name", Direction orderDirection = Direction.Ascending, string? name = null, string? editorUiAlias = null, string? editorAlias = null)
+        {
+            using ICoreScope scope = ScopeProvider.CreateCoreScope(autoComplete: true);
+
+            IEnumerable<IDataType> dataTypes = _dataTypeRepository.GetMany();
+
+            if (name is not null)
+            {
+                dataTypes = dataTypes.Where(datatype => datatype.Name?.Contains(name) ?? false);
+            }
+
+            if (editorUiAlias != null)
+            {
+                dataTypes = dataTypes.Where(datatype => datatype.EditorUiAlias?.Contains(editorUiAlias) ?? false);
+            }
+
+            if (editorAlias != null)
+            {
+                dataTypes = dataTypes.Where(datatype => datatype.EditorAlias.Contains(editorAlias));
+            }
+
+            ConvertMissingEditorsOfDataTypesToLabels(dataTypes);
+            scope.Complete();
+            return OrderByProperty(dataTypes, orderBy, orderDirection);
+        }
+
+        private IEnumerable<IDataType> OrderByProperty(IEnumerable<IDataType> dataTypes, string orderBy, Direction orderDirection)
+        {
+            // Get the property info for the specified property name
+            PropertyInfo? property = typeof(DataType).GetProperty(orderBy, BindingFlags.IgnoreCase | BindingFlags.Public | BindingFlags.Instance);
```

Comment:
I'm not a big fan of ordering by means of reflection. Maybe we should skip ordering entirely until we actually need it on the consumer side of the API?